Language improvements in the design document (#361)
luna/enso: doc/design/syntax/syntax.md

criteria, dependent types are much more flexible, they realize a continuum of
precision from the basic assertions we are used to expect from types up to a
complete specification of the program’s behaviour. It is the programmer’s choice

> to what degree to exploit the expressiveness of such a powerful type

Reviewer: ```suggestion
to what degree they want to exploit the expressiveness of such a powerful type
```

**If dependent types are so great, why they are not used widely?** Basically,
there are two problems. First, there is a small set of languages allowing for
dependent types, like Agda or Idris. Second, both writing as well as using

> dependently typed code is significantly harder than the code using conventional

Reviewer: ```suggestion
dependently typed code is significantly harder than for code using a conventional
```

> type system. The second problem is even bigger because it stands in a way to

Reviewer: ```suggestion
type system. The second problem is even bigger because it stands in the way of
```

> easily refactor the code base and keep it in a good shape.

Reviewer: ```suggestion
easily refactoring the code base and keeping it in good shape.
```

*** test.idr:18:23:unmatched case in Main.index ***
```

The above program crashed in the middle of execution. Such mistakes (related to the

> possibility of the index to be out of bounds) are very hard to catch and most

Reviewer: ```suggestion
possibility of the index being out of bounds) are very hard to catch, and most
```
